common/searchattribute/defs.go

```diff
@@ -279,6 +279,18 @@ func IsPreallocatedCSAFieldName(name string, valueType enumspb.IndexedValueType)
 	return re != nil && re.MatchString(name)
 }
 
+var chasmSearchAttributePattern = regexp.MustCompile(`^Temporal(Bool|Datetime|Int|Double|Keyword|KeywordList)(0[1-9]|[1-9][0-9])$`)
```

Comment:
I'd add the `Text` type here too, even though there are no `Text` type search attributes yet.